src/game/engine/template.cpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Chronoshift is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#include "template.h"
+#include "cell.h"
+#include "globals.h"
+#include "iomap.h"
+#include "tileset.h"
+
+#ifndef GAME_DLL
+TFixedIHeapClass<TemplateClass> g_Templates;
+#endif
+
+/**
+ * @brief
+ *
+ * 0x0056A1D8
+ */
+TemplateClass::TemplateClass(TemplateType type, cell_t cellnum) : ObjectClass(RTTI_TEMPLATE, g_Templates.ID(this))
+{
+    if (cellnum != -1) {
+        Unlimbo(Cell_To_Coord(cellnum));
+    }
+}
+
+/**
+ * @brief
+ *
+ * 0x0056A198
+ */
+void *TemplateClass::operator new(size_t size)
+{
+    TemplateClass *this_ptr = g_Templates.Alloc();
+    DEBUG_ASSERT(this_ptr != nullptr);
```

Comment:
You shouldn't really assert something that your function actually expects to happen and has a handler for. Asserts are for catching stuff that your function is written to never expect to happen.